packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -104,17 +117,139 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// This class stores the information of the selection under
+/// a [Selectable] or [SelectionHandler].
+///
+/// The [SelectedContentRange]s for a given [Selectable] or
+/// [SelectionHandler] can be retrieved by calling
+/// [SelectionHandler.getSelections].
+@immutable
+class SelectedContentRange {
+  /// Creates a [SelectedContentRange] with the given values.
+  const SelectedContentRange({
+    required this.contentLength,
+    required this.contentStart,
+    required this.startOffset,
+    required this.endOffset,
+  });
+
+  /// A selected content range that contains nothing.
+  static SelectedContentRange empty({int? contentLength}) {
+    return SelectedContentRange(
+      startOffset: -1,
+      endOffset: -1,
+      contentStart: -1,
+      contentLength: contentLength ?? -1,
+    );
+  }
+
+  /// The length of the content.
+  ///
+  /// The absolute value of the difference between the start
+  /// offset and end offset contained by this [SelectedContentRange]
+  /// must not exceed the content length.
+  final int contentLength;
+
+  /// The offset where the content begins.
+  final int contentStart;
+
+  /// The start of the selection relative to the start of the content.
+  ///
+  /// {@template flutter.rendering.selection.SelectedContentRange.selectionOffsets}
+  /// For example a [Text] widget's content is in the format
+  /// of an [InlineSpan] tree.
+  ///
+  /// Take the [Text] widget and [TextSpan] tree below:
+  ///
+  /// {@tool snippet}
+  /// ```dart
+  /// const Text.rich(
+  ///   TextSpan(
+  ///     text: 'Hello world, ',
+  ///     children: <InlineSpan>[
+  ///       WidgetSpan(
+  ///         child: Text('how are you today?'),
+  ///       ),
+  ///     ],
+  ///   ),
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// If we select from the beginning of 'world' to the
+  /// end of the '?' in the [WidgetSpan], we will receive
+  /// two [SelectedContentRange]s from [SelectionHandler.getSelections].
+  /// The first range will be relative to the root text
+  /// of the [TextSpan], the [startOffset] will be 6,
+  /// and [endOffset] will be 13. The second range
+  /// will be relative to the content inside the
+  /// [WidgetSpan], the [startOffset] will be 0,
```

Comment:
Why not use global content length? and selection start also in global offset.

The current implementation will tie us into how text are broke into piece. If we changes it in the future the the select


if we use relative content range, they have to know the widget subtree and also how text will be break into multiple selectable. It will also change SelectedContentRange

Reply:
I see the concern. I think this is cleared up now, with returning one SelectedContentRange instead of a list.